packages/deployer/utils/deployments.js

```diff
@@ -0,0 +1,69 @@
+const fs = require('fs');
+const path = require('path');
+const glob = require('glob');
+const naturalCompare = require('string-natural-compare');
+const relativePath = require('./relative-path');
+
+// Regex for deployment file formats, e.g.: 2021-08-31-00-sirius.json
+const DEPLOYMENT_FILE_FORMAT = /^[0-9]{4}-[0-9]{2}-[0-9]{2}-[0-9]{2,}(?:-[a-z0-9]+)?\.json$/;
+
+function getMainProxyAddress(instance = 'official') {
+  const data = getDeploymentData(instance);
+
+  return data.contracts['contracts/MainProxy.sol'].deployedAddress;
+}
+
+function getDeploymentData(instance = 'official') {
+  const file = getDeploymentFile(instance);
+
+  if (!file) {
+    throw new Error(
+      `No deployment file found on network "${hre.network.name}" for instance "${instance}".`
+    );
+  }
+
+  return JSON.parse(fs.readFileSync(file, 'utf8'));
+}
+
+function getDeploymentFile(instance = 'official') {
+  const deployments = getDeploymentFiles(instance);
+
+  return deployments.length > 0 ? deployments[deployments.length - 1] : null;
+}
+
+function getDeploymentFiles(instance = 'official') {
+  const paths = getDeploymentPaths(instance);
+  const folder = paths.instance;
+
+  return glob
+    .sync(`${folder}/*.json`)
+    .filter((file) => DEPLOYMENT_FILE_FORMAT.test(path.basename(file)))
+    .sort(naturalCompare);
+}
+
+function getDeploymentPaths(instance = 'official') {
+  const paths = {};
+
+  paths.deployments = path.resolve(hre.config.paths.root, hre.config.deployer.paths.deployments);
+  paths.modules = path.resolve(hre.config.paths.root, hre.config.deployer.paths.modules);
+  paths.network = path.join(paths.deployments, hre.network.name);
+  paths.instance = path.join(paths.network, instance);
+  paths.extended = path.join(paths.instance, 'extended');
+  paths.routerTemplate = path.resolve(__dirname, '../templates/GenRouter.sol.mustache');
+  paths.routerPath = relativePath(
+    path.join(hre.config.paths.sources, `${hre.deployer.routerModule}.sol`)
+  );
+  paths.proxyPath = relativePath(
+    path.join(hre.config.paths.sources, `${hre.config.deployer.proxyName}.sol`)
+  );
+
+  return paths;
+}
```

Comment:
As a response to the PR's description question, Yes, it would be great if we do these utils to not use the `hre` global variable on context. We could plan a refactor to change all the interfaces of these function to take everything they need as parameters, so they will be easily reusable from the tests.

E.g., this is an ugly example, which could be improved, but is 100% reusable without the global context:

```javascript
function getDeploymentPaths({
  instance,
  routerModule,
  proxyName,
  network,
  root,
  sources,
  deployments,
  modules,
}) {
  const paths = {};

  paths.deployments = path.resolve(root, deployments);
  paths.modules = path.resolve(root, modules);
  paths.network = path.join(paths.deployments, network);
  paths.instance = path.join(paths.network, instance);
  paths.extended = path.join(paths.instance, 'extended');
  paths.routerTemplate = path.resolve(__dirname, '../templates/GenRouter.sol.mustache');
  paths.routerPath = relativePath(
    path.join(sources, `${routerModule}.sol`)
  );
  paths.proxyPath = relativePath(
    path.join(sources, `${proxyName}.sol`)
  );

  return paths;
}
```

Then, call it once at the beginning of the task:
```javascript
hre.deployer.paths = getDeploymentPaths({
  instance: 'official',
  routerModule: 'GenRouterLocalOfficial',
  proxyName: hre.config.deployer.proxyName,
  network: hre.network.name,
  root: hre.config.paths.root,
  sources: hre.config.paths.sources,
  deployments: hre.config.deployer.paths.deployments,
  modules: hre.config.deployer.paths.modules,
})

```

Reply:
This would make the code a little harder to track all the variables we send around, but more explicit, it would not require the reader to know about global context, and easier to reuse on tests, etc